pkg/util/command_util.go

```diff
@@ -469,3 +469,11 @@ func getUID(userStr string) (uint32, error) {
 	}
 	return uint32(uid), nil
 }
+
+// ExtractFilename extracts the filename from a URL without its query url
+func extractFilename(rawURL string) string {
+	// not necessarily return an error, due to parsing ambiguities https://pkg.go.dev/net/url#Parse
+	parsedURL, _ := url.Parse(rawURL)
```

Comment:
I see the comment above but can you explain why we don't surface or bubble up this error?   From the url.Parse docs

```
// The url may be relative (a path, without a host) or absolute
// (starting with a scheme). Trying to parse a hostname and path
// without a scheme is invalid but may not necessarily return an
// error, due to parsing ambiguities.
```

I believe we should bubble up this error, let me know if that makes sense or there is some issue that occurs when doing this.  I believe the COPY value when in this method `URLDestinationFilepath` should have a scheme so we should respect the error

Reply:
Hi @aaron-prindle 

Thanks, I thought that the filepath example.com/thefile.txt would return an error since it doesn't have the scheme, but it actually doesn't. Thanks for the comment. I've already fixed it and switched the execution order, with ResolveEnvironmentReplacement being executed before extractFilename, i think thats better to replace the env first.

kindly check